### ISSUE TITLE
Sidebar right for mobile should close after option selection

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#711]
### Terms

- [X] I have searched all [open bug reports](https://github.com/activist-org/activist/issues?q=is%3Aopen+is%3Aissue+label%3Abug)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Behavior

In conversations with @lucianmurmurache, SidebarRight is not closing when an option is selected. This can be seed from the mobile view on any page where the burger menu is used to open the sidebar, and then once a new localization, color mode or another option is selected the sidebar is still open. 

Suggestion for how to fix this would be:
> A ref and a listener for click outside should do the trick here

This is a good first issue for someone wanting to help, or a community member can also pick it up 😊

**Comment by lucianmurmurache** [comment]
Happy to take care of it @andrewtavis 

There is another [issue ](https://github.com/activist-org/activist/issues/631)that could potentially reuse the same logic

**Comment by lucianmurmurache** [comment]
@andrewtavis I am actually not 100% sure if the functionality would be sound from an UX perspective, do you know if there is any community member that specialises in UX to give me a little hint here? 

Considering that there are two menu items, which themselves are drop downs, I'm thinking that it might be less convenient if the user wants to change both language and theme, they would then have to open the menu twice. I can take care of the menu closure on click outside for now and leave this as a draft until we get some clarity 💡

**Comment by andrewtavis** [comment]
I think clicking outside would be fine for now, @lucianmurmurache :) We don't have a specific UX person right now, but we can also chat about it in the next dev sync 😊

**Comment by lucianmurmurache** [comment]
In that case, the PR is ready @andrewtavis 👍

**Comment by andrewtavis** [comment]
a536fc6 and a minor change in 3e9aec5 do some final touches for this, @lucianmurmurache :) One thing I noticed post merge is that it was opening on any click outside, so even a random click on the page would open the sidebar 😅 We now have `closeMenuState` that just sets it to false that's called. There likely is doubtless a way of doing this, but I wanted to fix prod quickly (hotfix welcome as we're doubtless calling this TS function needlessly when the sidebar is closed). 

Beyond this we now have [SidebarToggle](https://github.com/activist-org/activist/blob/main/frontend/components/sidebar/SidebarToggle.vue) that houses the logic for the original left sidebar toggle, and this is now also used for the right sidebar, but just reversed. I also removed your close button as I felt that for now it's looking ok if we leave it in the same place as before and pop it to the top with a `z-40` and the sidebar at `z-30`, which actually was why we lost the old close icon. I've been reducing z-indexes to make sure things aren't above the load screen that's at `z-50`, but then the close icon was then beneath the sidebar. All for the best as the consistency is really great 😊

Feedback welcome! Hope that everything's looking/working alright :)
